Taskfile.yml

```diff
@@ -0,0 +1,163 @@
+# https://taskfile.dev/
+
+version: '3'
+
+vars:
+  BINARY_PATH: "${HOME}/.terraform.d/plugins/registry.terraform.io/opslevel/opslevel"
+  OPSLEVEL_DIR: "{{.ROOT_DIR}}/opslevel"
+  VERSION_DYNAMIC:
+    sh: echo "$(git describe --tags --abbrev=0 | tr -d 'v')$(date +%s)"
+  WORKSPACE_DIR: "{{.ROOT_DIR}}/workspace"
+
+tasks:
+
+  install-changie:
+    desc: Install changie
+    aliases: [get-changie, get-changie]
+    cmds:
+      - echo "Installing changie..."
+      - task: install-changie-mac-os
+      - echo -e "\nSuccess! 'terraform' installed."
+
+  install-changie-mac-os:
+    internal: true
+    platforms: [darwin]
+    cmds:
+      - brew tap miniscruff/changie https://github.com/miniscruff/changie
+      - brew install changie
+    preconditions:
+      - sh: 'which brew'
+        msg: '"brew" needed to install changie - see https://brew.sh'
+
+  install-terraform:
+    desc: Install terraform
+    aliases: [get-tf, get-terraform]
+    cmds:
+      - echo "Installing terraform..."
+      - task: install-terraform-mac-os
+      - echo -e "\nSuccess! 'terraform' installed."
+    silent: true
+
+  install-terraform-mac-os:
+    internal: true
+    platforms: [darwin]
+    cmds:
+      - brew tap hashicorp/tap
+      - brew install hashicorp/tap/terraform
+    preconditions:
+      - sh: 'which brew'
+        msg: '"brew" needed to install terraform - see https://brew.sh'
+
+  lint:
+    desc: Formatting and linting go code
+    dir: "{{.OPSLEVEL_DIR}}"
+    cmds:
+      - gofmt -d .
+      - go vet ./...
+      - golangci-lint run
+
+  lintfix:
+    desc: Fix formatting and linting
+    dir: "{{.OPSLEVEL_DIR}}"
+    env:
+      TF_ACC: true
+    cmds:
+      - gofmt -w .
+      - go mod tidy
+      - golangci-lint run --fix
+
+  terraform-apply:
+    desc: Run terraform apply
+    aliases: [tf-apply]
+    dir: "{{.WORKSPACE_DIR}}"
+    cmds: [terraform apply]
+    preconditions:
+      - sh: 'which terraform'
+        msg: '"terraform" needed - run "brew install terraform"'
+
+  terraform-build:
+    desc: Build terraform
+    aliases: [tf-build]
+    platforms: [darwin]
+    vars:
+      BINARY: terraform-provider-opslevel_{{.VERSION_DYNAMIC}}
+      VERSION:
+        sh: echo "{{.BINARY}} | cut -d '_' -f 2"
+    cmds:
+      - go build -o {{.BINARY}} || exit 1
+      - chmod +x {{.BINARY}}
+      - mkdir -p {{.BINARY_PATH}}
+      - mv {{.BINARY}} {{.BINARY_PATH}}
+      - cmd: echo "Built terraform provider at:"
+        silent: true
+      - cmd: echo -e "  {{.BINARY_PATH}}/{{.BINARY}}"
+        silent: true
+
+  terraform-clean:
+    desc: Clean up workspace
+    aliases: [tf-clean]
+    cmds:
+      - rm -rf {{.BINARY-PATH}}
```

Comment:
whats the point of this?  should we also cleanup the workspace directory?

additionally one thing that i often do for cleanup is delete some of the other files that terraform generates in the workspace directory - basically everything that is not `backend.tf` and `main.tf`

Reply:
For easy reference:
`BINARY_PATH: "${HOME}/.terraform.d/plugins/registry.terraform.io/opslevel/opslevel"`

I just copy + pasted what I saw in the original [clean script ](https://github.com/OpsLevel/terraform-provider-opslevel/blob/80eb044483b5e62aa44c9a9cd195e25539aa795f/clean). This can definitely be improved on.